src/librustc_mir/transform/check_consts/ops.rs

```diff
@@ -183,6 +183,10 @@ impl NonConstOp for Loop {
 #[derive(Debug)]
 pub struct MutBorrow(pub BorrowKind);
 impl NonConstOp for MutBorrow {
+    fn feature_gate(tcx: TyCtxt<'_>) -> Option<bool> {
+        Some(tcx.features().const_mut_refs)
```

Comment:
This feature should only work for `BorrowKind::Mut`. We don't want to allow arbitrary interior mutability with this feature gate. I think you have to split the `MutBorrow` feature into `MutBorrow` and `CellBorrow` (both without the `BorrowKind` field, and then dispatch at the `MutBorrow` use sites depending on the `BorrowKind`.